packages/webpack-plugin/src/runtime.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-var */
+type StateValue = boolean | number | string;
+
+interface StateMap {
+    [stateName: string]: StateValue;
+}
+
+interface Host {
+    sts: any;
+    stc: any;
+    sti: any;
+}
+
+export function stylesheet(host: Host) {
+    var stateMiddleDelimiter = '-';
+    var booleanStateDelimiter = '--';
+    var stateWithParamDelimiter = '---';
+
+    function createBooleanStateClassName(namespace: string, stateName: string) {
+        return namespace + booleanStateDelimiter + stateName;
+    }
+
+    function createStateWithParamClassName(namespace: string, stateName: string, param: string) {
+        return (
+            namespace +
+            stateWithParamDelimiter +
+            stateName +
+            stateMiddleDelimiter +
+            param.length +
+            stateMiddleDelimiter +
+            param.replace(/\s/gm, '_')
+        );
+    }
+
+    function createStateClass(
+        namespace: string,
+        stateName: string,
+        stateValue: StateValue
+    ): string {
+        if (
+            stateValue === false ||
+            stateValue === undefined ||
+            stateValue === null ||
+            stateValue !== stateValue // check NaN
+        ) {
+            return '';
+        }
+
+        if (stateValue === true) {
+            // boolean state
+            return createBooleanStateClassName(namespace, stateName);
+        }
+
+        var valueAsString = stateValue.toString();
+
+        return createStateWithParamClassName(namespace, stateName, valueAsString);
+    }
+
+    function style(namespace: string) {
+        var classNames = [];
+
+        for (var i = 1; i < arguments.length; i++) {
+            // eslint-disable-next-line prefer-rest-params
+            var item = arguments[i];
+
+            if (item) {
+                if (typeof item === 'string') {
+                    classNames[classNames.length] = item;
+                } else if (i === 2) {
+                    for (var stateName in item) {
+                        var stateValue = item[stateName];
```

Comment:
hasownproperty?

Reply:
not really needed I don't expect here usage of complex inherited classes here.